src/DependencyInjection/Configuration.php

```diff
@@ -768,6 +769,12 @@ private function createCachePluginNode()
                         ->end()
                     ->end()
                 ->end()
+                ->arrayNode('cache_listeners')
+                    ->info('A list of service ids to act on the response based on the results of the cache check. Must implement '.CacheListener::class.'. Defaults to an empty array.')
+                    ->beforeNormalization()->ifNull()->thenEmptyArray()->castToArray()->ifEmpty()->thenUnset()->end()
```

Comment:
i am still confused about `ifEmpty()->thenUnset()`. could it be that this will break in the extension if i configure the cache plugin but no listeners? although the "empty" configuration test seems to see the array, but then i feel like that unset does not do anything ever?

Reply:
uhmmm... yea, this line is kind of f***ed up. ended up into just casting it to array, which should be fine.